x-pack/plugins/spaces/public/spaces_manager/spaces_manager.ts

```diff
@@ -30,8 +30,13 @@ export class SpacesManager {
     this.refreshActiveSpace();
   }
 
-  public async getSpaces(purpose?: GetSpacePurpose): Promise<Space[]> {
-    return await this.http.get('/api/spaces/space', { query: { purpose } });
+  public async getSpaces(
+    purpose?: GetAllSpacesPurpose,
+    allowPartialAuthorization?: boolean
```

Comment:
Can we update this signature to match the one we land on for the server-side `SpacesClient`? The object-based approach we took there is easier to read IMO, especially when dealing with these optional boolean parameters

Reply:
Done in 836ef49!